src/main/java/com/programmers/VoucherManagementApplication/repository/JdbcVoucherRepository.java

```diff
@@ -0,0 +1,95 @@
+package com.programmers.VoucherManagementApplication.repository;
+
+import com.programmers.VoucherManagementApplication.vo.Amount;
+import com.programmers.VoucherManagementApplication.vo.VoucherType;
+import com.programmers.VoucherManagementApplication.voucher.Voucher;
+import org.springframework.context.annotation.Profile;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.nio.ByteBuffer;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Repository
+@Profile("jdbc")
+public class JdbcVoucherRepository implements VoucherRepository {
+
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    private static final RowMapper<Voucher> voucherRowMapper = (resultSet, i) -> {
+        UUID voucherId = toUUID(resultSet.getBytes("voucher_id"));
+        String type = resultSet.getString("voucher_type"); //fixed_discount vs fixed
+        long amount = resultSet.getLong("amount");
+        return Voucher.from(voucherId, VoucherType.getName(type), new Amount(amount));
+    };
+
+    public JdbcVoucherRepository(NamedParameterJdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private Map<String, Object> toParamMap(Voucher voucher) {
+        return new ConcurrentHashMap<>() {{
+            put("voucherId", voucher.getVoucherId().toString().getBytes());
+            put("voucherType", String.valueOf(voucher.getVoucherType()));
+            put("amount", voucher.getAmount().getAmount());
+            put("usageStatus", voucher.getUsageStatus());
+        }};
+    }
```

Comment:
```suggestion
    private Map<String, Object> toParamMap(Voucher voucher) {
        return Map.of(
	        VOUCHER_ID, voucher.getVoucherId().toString().getBytes()
	        , VOUCHER_TYPE, String.valueOf(voucher.getVoucherType())
	        , AMOUNT, voucher.getAmount().getValue()
	        , USAGE_STATUS, voucher.getUsageStatus()
        );
    }
```

동일 합니다. Map.of 한번 사용해보시죠!


```java
	public static final String VOUCHER_ID = "voucher_id";
	public static final String VOUCHER_TYPE = "voucher_type";
	public static final String AMOUNT = "amount";
	public static final String USAGE_STATUS = "usageStatus";
```